### PR TITLE
Feature/connect to api

### PR DESCRIPTION
**Pull request by Braingenius2**
# Connect to API

This pull request connects the bookstore application to an existing API to create and remove books from a server. 

## Changes
The following changes were implemented:

- [x] Made use of the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as backend
- [x] Used [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make HTTP requests to the backend
- [x] Fetched books from the Bookstore API after the component has rendered
- [x] Refactored add book and remove book features to persist changes in the server

### CODE REVIEW

**Review by anmol111pal — APPROVED** [review]
Great work !

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -21,22 +22,78 @@ const initialState = {
...
+});
+
+const addBook = createAsyncThunk('books/addBook', async (newBook, { rejectWithValue }) => {
+  try {
+    await axios.post(`${baseUrl}/apps/${appId}/books`, newBook);
+    return newBook;
+  } catch (error) {
+    return rejectWithValue('Something went wrong with adding new book', error);
+  }
+});
```
Comment (mikethreels) [inline src/redux/books/booksSlice.js:33]: - [ ] when a user adds a book and then reloads the page the book appears twice, please fix this before requesting another review.

File: src/redux/books/booksSlice.js [hunk 2]
```diff
@@ -21,22 +22,78 @@ const initialState = {
...
+
+const removeBook = createAsyncThunk('books/removeBook', async (itemId, { rejectWithValue }) => {
+  try {
+    await axios.delete(`${baseUrl}/apps/${appId}/books/${itemId}`);
+    return itemId;
+  } catch (error) {
+    return rejectWithValue('Something went wrong with removing book', error);
+  }
+});
```
Comment (mikethreels) [inline src/redux/books/booksSlice.js:42]: - [ ] when a user removes a book and then reloads the page the book appears again,  please fix this before requesting another review.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Fortune, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Good job connecting to the API

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — APPROVED** [review]
# Hi @Braingenius2,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  No Linters errors 🔥
- [ ]  Great job working on functionality for connecting the API 💯
- [ ]  Followed GitFlow ⭐🚀
- [ ] Redux set up well with the API 💪🏻
- [ ] Descrtipive PR⭐



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**